tensorflow_federated/python/core/impl/executors/federating_executor.py

```diff
@@ -453,6 +454,353 @@ def validate_executor_placements(cls, executor_placements):
     raise NotImplementedError('The secure sum intrinsic is not implemented.')
 
 
+class TrustedAggregatorIntrinsicStrategy(IntrinsicStrategy):
+
+  def __init__(self, federating_executor):
+    super().__init__(federating_executor)
+
+  @classmethod
+  def validate_executor_placements(cls, executor_placements):
+    singleton_placements = [
+        placement_literals.SERVER,
+        placement_literals.AGGREGATOR,
+    ]
+    expected_placements = singleton_placements + [placement_literals.CLIENTS]
+    py_typecheck.check_type(executor_placements, dict)
+    for k, v in executor_placements.items():
+      if k is not None:
+        py_typecheck.check_type(k, placement_literals.PlacementLiteral)
+      py_typecheck.check_type(v, (list, executor_base.Executor))
+      if isinstance(v, list):
+        for e in v:
+          py_typecheck.check_type(e, executor_base.Executor)
+      for pl in expected_placements:
+        if pl not in executor_placements:
+          raise ValueError(
+              'Improper executor placements, expected executor stack for '
+              '{} placement.'.format(pl.name))
+      for pl in [None] + singleton_placements:
+        if pl in executor_placements:
+          ex = executor_placements[pl]
+          if isinstance(ex, list):
+            pl_cardinality = len(ex)
+            if pl_cardinality != 1:
+              raise ValueError(
+                  'Unsupported cardinality for placement "{}": {}.'.format(
+                      pl, pl_cardinality))
+
+  async def _trusted_aggregator_generate_keys(self):
+
+    @computations.tf_computation()
+    def generate_keys():
+      pk, sk = easy_box.gen_keypair()
+      return pk.raw, sk.raw
+
+    fn_type = generate_keys.type_signature
+    fn = generate_keys._computation_proto
+
+    aggregator_ex = self._get_child_executors(
+        placement_literals.AGGREGATOR, index=0)
+
+    result = await aggregator_ex.create_call(await aggregator_ex.create_value(
+        fn, fn_type))
+
+    keys = await asyncio.gather(*[
+        aggregator_ex.create_selection(result, i)
+        for i in range(len(result.type_signature))
+    ])
+
+    pk_fed_val, sk_fed_val = await asyncio.gather(
+        *[self._place(k, placement_literals.AGGREGATOR) for k in keys])
+
+    pk_fed_val = await self.federated_broadcast(pk_fed_val)
+
+    return pk_fed_val, sk_fed_val
+
+  async def _zip_val_key(self, val, key):
+
+    # zip values and keys EagerValues on each client using
+    # their respective eager executor.
+    val_key_zips = []
+    key_placement = key.type_signature.placement.name
+    for i in range(len(val)):
+      client_ex = self._get_child_executors(placement_literals.CLIENTS, index=i)
+      val_eager_val = val[i]
```

Comment:
Are we dependent on these being `EagerValues`? It doesn't seem like we are, so maybe it's just a rename that's needed here. Ideally we should not make assumptions about what type of executor stack is placed on CLIENTS